tests/evaluate/test_default_evaluator.py

```diff
@@ -187,7 +187,8 @@ def test_regressor_evaluation(
 
     expected_metrics = _get_regressor_metrics(y, y_pred, sample_weights=sample_weights)
     expected_metrics["score"] = model._model_impl.score(
-        diabetes_dataset.features_data, diabetes_dataset.labels_data, sample_weight=sample_weights
+        diabetes_dataset.features_data,
+        params={"y": diabetes_dataset.labels_data, "sample_weight": sample_weights},
```

Comment:
is this a breaking change?

Reply:
I think no, `_model_impl` is not user facing API.

Reply:
but it's marked as `developer_stable`

Reply:
Yes. Does `developer_stable` this matters ? this is for developers. @BenWilson2 ?

Reply:
It does. It would affect developers, right?

Reply:
Then we need to reconsider how to fix the issue. Current approach inevitably has to break this.